torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -182,12 +214,16 @@ class _OverlapInfo():
     """
     def __init__(self) -> None:
         self.status: _OverlapStatus = _OverlapStatus.UNINITIALIZED
+        self.uniform_bucket_assignment: bool = False
 
+        world_size = dist.get_world_size()
```

Comment:
I originally factored this out since I also used it when initializing another data structure, which has now been removed. I will get rid of `world_size` and pass `dist.get_world_size()` directly to `range()` on line 223.